openml/extensions/sklearn/extension.py

```diff
@@ -1646,20 +1673,28 @@ def _prediction_to_probabilities(y: np.ndarray, classes: List[Any]) -> np.ndarra
                     # then we need to add a column full of zeros into the probabilities
                     # for class 3 because the rest of the library expects that the
                     # probabilities are ordered the same way as the classes are ordered).
-                    proba_y_new = np.zeros((proba_y.shape[0], len(task.class_labels)))
-                    for idx, model_class in enumerate(model_classes):
-                        proba_y_new[:, model_class] = proba_y[:, idx]
-                    proba_y = proba_y_new
-
-                if proba_y.shape[1] != len(task.class_labels):
                     message = "Estimator only predicted for {}/{} classes!".format(
                         proba_y.shape[1], len(task.class_labels),
                     )
                     warnings.warn(message)
                     openml.config.logger.warn(message)
+
+                    # True if predict_proba is successfully called with X_test as numpy
+                    # if X_test is dataframe, proba_y will be dataframe with model_classes columns
+                    if isinstance(proba_y, np.ndarray):
+                        proba_y = pd.DataFrame(proba_y, columns=model_classes)
+
+                    for i, col in enumerate(task.class_labels):
+                        # adding missing columns with 0 probability
+                        if col not in model_classes:
+                            proba_y[col] = 0
```

Comment:
Don't you have to re-order the columns if you add a new column to the dataframe?

Reply:
Doesn't the next line, at the same scope as the `for` do that?

Reply:
My bad, of course it does.